### ISSUE TITLE
How to deal with url/_search?

### ISSUE THREAD

**Issue by teebu** [sscarduzio/elasticsearch-readonlyrest-plugin#111]
I know we're setting permission for each index, but this bypasses everything, and returns all indices. 

From this, you can do any query you want. 

`_search?q=message:hellotest1` for example, will give you data for all indices with that field, even if its not in the allowed ones.

**Comment by sscarduzio** [comment]
Thanks for reporting this @teebu.

Just tried how the plugin sees this use case:
```
checking request:{ action: indices:data/read/search, OA:172.17.0.1, indices:[], M:GET, P:/_search, C:, Headers:[]}
```
looks like no indices are associated with this action, so looks like we're touching no indices at all (rather than all of them).

I'm starting to believe this approach of inbound index checks in requests is quite limited, ideally we'd need to (also) analyse/filter the results of the queries. Easier said than done, but I'll have a look.

For now, you can patch the hole with a `url_re` rule, or forbidding the access to the index `<no-index>`

**Comment by teebu** [comment]
A general forbid may be enough.

```
    - name: forbid no index access
      type: forbid
      indices: ["<no-index>"]
```

Then I remembered about the explicit index stuff that ES requires you to disable for logstash to work.

**Comment by sscarduzio** [comment]
Ok @teebu, I got a possible solution.

I'm able to detect if the request at hand is of type "SearchRequest" and if I see it has no reference to a particular index (like the generic `/_search` above). In this case, instead of parsing it as `<no-index>`, I parse it as "_all" index.
 
If one really wants this all indices search to work, you need to add "_all" as allowed index. 

A phase two will convert the absence of indices in a search request as "give me all the indices I'm allowed to see", but this requires #110 .

Will this work for you?

**Comment by sscarduzio** [comment]
https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/releases/tag/v1.12.0-rc1_es-v2.3.3

**Comment by teebu** [comment]
So, it will block `_all` by default, since it's not on the allowed list, im ok with that.

**Comment by sscarduzio** [comment]
https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/releases/tag/v1.12.0-rc1_es-v2.4.1

**Comment by sscarduzio** [comment]
this is addressed in 1.12.1
